x/liquidstake/keeper/liquidstake.go

```diff
@@ -719,6 +720,54 @@ func (k Keeper) LiquidUnbond(
 	return completionTime, returnAmount, ubd, nil
 }
 
+// DivideByWeight divide the input value by the ratio of the param weight of the liquid validator and return it with crumb
+// which is may occur while dividing according to the weight of active liquid validators by decimal error.
+func (k Keeper) DivideByWeight(
+	ctx sdk.Context,
+	avs types.ActiveLiquidValidators,
+	input math.Int,
+	whitelistedValsMap types.WhitelistedValsMap,
+) (outputs []math.Int, crumb math.Int) {
+	totalWeight := avs.TotalWeight(whitelistedValsMap)
+	if !totalWeight.IsPositive() {
+		return []math.Int{}, sdk.ZeroInt()
+	}
+
+	totalOutput := sdk.ZeroInt()
+	unitInput := math.LegacyNewDecFromInt(input).QuoTruncate(math.LegacyNewDecFromInt(totalWeight))
+	for _, val := range avs {
+		validator, _ := k.stakingKeeper.GetValidator(ctx, val.GetOperator())
+
+		// calculate the shares the input would receive
+		output := unitInput.MulInt(val.GetWeight(whitelistedValsMap, true)).TruncateInt()
+		outputShares := validator.GetDelegatorShares().MulInt(output).QuoInt(validator.GetTokens())
+
+		// just delegate if the validator does not exceed any of the validator specific caps
+		if !k.stakingKeeper.CheckExceedsValidatorBondCap(ctx, validator, outputShares) &&
+			!k.stakingKeeper.CheckExceedsValidatorLiquidStakingCap(ctx, validator, outputShares, false) {
```

Comment:
here if it still has some space, we should try to fill everything up

Reply:
what do you mean ?